draft-ietf-ppm-dap.md

```diff
@@ -710,6 +710,163 @@ last two weeks of the current time) and reject reports that fall outside of the
+║  ║  .-+-+------------+-+------+.     ║   ║       ║       |       ║   ║       ║
+║  ║ |    prepare input shares    |<---║---║-------║-------|       ║   ║       ║
+║  ║  '---+-------------+--------'     ║   ║       ║       |       ║   ║       ║
+║  ║      |             |              ║   ║       ║       |       ║   ║       ║
+║  ║      v             v              ║   ║<------║-------|       ║   ║       ║
+║  ║  Leader output   Helper output    ║   ║       ║       |       ║   ║       ║
+║  ║    share 1         share 1        ║   ║       ║      ...      ║   ║       ║
+║  +=======|===============|===========+   ║       ║               ║   ║       ║
+║    ║     | Leader output | Helper output ║       ║       |       ║   ║       ║
+║    ║     | share 2       | share 2       ║       ║<------'       ║   ║       ║
+║    +=====|=|=============|=|=============+       ║               ║   ║       ║
+║      ... | |  Leader     | | Helper              ║               ║   ║       ║
+║          | |  output     | | output              ║               ║   ║       ║
+║       ║  | |  share k    | | share k             ║               ║   ║       ║
+║       +==|=|=====|=======|=|=====|===============+               ║   ║       ║
+║          v v ... v       v v ... v                               ║   ║       ║
+║  .-------+-+-----+-.   .-+-+-----+-------.                       ║   ║       ║
+║ | accumulate leader | | accumulate helper |                      ║   ║       ║
+║ |   output shares   | |   output shares   |                      ║   ║       ║
+║  '----+------------'   '-------------+---'                       ║   ║       ║
+║       |                              |                           ║   ║       ║
+║       v                              v                           ║   ║       ║
+║  Leader batch                    Helper batch                    ║   ║       ║
+║    bucket 1                        bucket 1                      ║   ║       ║
++=======|===============================|==========================+   ║       ║
+  ║     | Leader batch                  | Helper batch                 ║       ║
+  ║     |   bucket 2                    |  bucket 2                    ║       ║
+  +=====|========|======================|=====|========================+       ║
+   ...  | .------'                      | .---'                                ║
+        | |  Leader batch               | |  Helper batch                      ║
+    ║   | |    bucket j                 | |    bucket j                        ║
+    +===|=|=====|=======================|=|=====|==============================+
+        v v ... v      query chosen     v v ... v
+     .--+-+-----+--.   by collector  .--+-+-----+--.
+    |  merge leader |       |       | merge helper  |
+    | batch buckets |<------+------>| batch buckets |
+     '------+------'                 '------+------'
+            |                               |
+            '---------------+---------------'
```

Comment:
Perhaps we should show aggregate shares as intermediate values here.

Reply:
Oops, you're absolutely right. Those got lost when I reorganized the diagram.